src/renderer/PresetsBar.tsx

```diff
@@ -152,66 +105,110 @@ const PresetsBar = () => {
       try {
         await deletePreset(deletedValue);
+        setPresetNames(presetNames.filter((n) => n !== deletedValue));
       } catch (e) {
         // continue to run, the worst case is that the file still exists and that's all.
       }
-    };
+    },
+    [presetNames]
+  );
+
+  // Renaming an existing preset
+  const handleRenameExistingPresetName = useCallback(
+    (oldValue: string) => async (newValue: string) => {
+      try {
+        await renamePreset(oldValue, newValue);
+        setPresetNames(
+          // Keep presets sorted
+          presetNames.map((n) => (n === oldValue ? newValue : n)).sort()
+        );
+      } catch (e) {
+        setGlobalError(e as ErrorDescription);
+      }
+    },
+    [presetNames, setGlobalError]
+  );
 
+  // Validating a new preset name
+  const validatePresetName = useCallback((newValue: string) => {
+    if (RESERVED_FILE_NAMES_SET.has(newValue.toUpperCase())) {
+      return 'Invalid preset name, please use another.';
+    }
+
+    return undefined;
+  }, []);
+
+  // Validating a preset rename
+  const validatePresetRename = useCallback(
+    (newValue: string) => {
+      if (!newValue) {
+        return 'Preset name cannot be empty.';
+      }
+      if (
+        presetNames.some(
+          (oldValue) => newValue.toLowerCase() === oldValue.toLowerCase()
+        )
+      ) {
+        return 'Duplicate name found.';
+      }
+
+      return validatePresetName(newValue);
+    },
+    [validatePresetName, presetNames]
+  );
+
+  const options: IOptionEntry[] = useMemo(() => {
     return presetNames.map((n) => {
       return {
         value: n,
         label: n,
         display: (
           <PresetListItem
             value={n}
-            handleChange={(newValue: string) =>
-              handleEditExistingPresetName(n, newValue)
-            }
-            handleDelete={handleDeletePreset}
+            handleChange={handleRenameExistingPresetName(n)}
+            handleDelete={handleDeletePreset(n)}
             isDisabled={!!globalError}
+            validate={validatePresetRename}
           />
         ),
       };
     });
-  }, [globalError, presetNames]);
+  }, [
+    validatePresetRename,
+    globalError,
+    handleDeletePreset,
+    handleRenameExistingPresetName,
+    presetNames,
+  ]);
 
   return (
     <div className="presetsBar">
       <h4>Preset Menu</h4>
-      <div className="row center presetName">
-        Name:&nbsp;
+      <div className="row">
+        <div className="presetName">Name:&nbsp;</div>
         <TextInput
           value={presetName}
           ariaLabel="Preset Name"
           isDisabled={!!globalError}
           handleChange={handleChangeNewPresetName}
+          formatInput={formatPresetName}
+          validate={validatePresetName}
         />
       </div>
       <Button
         ariaLabel="Save settings to preset"
-        className="small full"
+        className="small"
         isDisabled={!!globalError || !presetName}
```

Comment:
We probably want to disable if someone has an invalid preset name typed in. I think we should just pass TextInput a set error function?

Reply:
Refactored `TextInput` so that validation and managing the error message is handled outside of the component - giving us better control regarding when the validation should occur. Thoughts? @xenown 

Currently there's 2 different `TextInput` use cases:
1. Validate and update the value on change
   - ex: input for saving a preset
   - errors will appear as you type, and making a subsequent change will automatically revalidate the input
2. Validate and update the value on submit (hitting enter)
   - ex: inputs for renaming a preset
   - errors only appear on submit - thus errors will continue to be present until a subsequent submission
   
Thoughts on these UX patterns? Should we make them more consistent? If yes, which pattern is preferred?
@h39s @flyingpandaboy @xenown